app/controllers/application_controller.rb

```diff
@@ -35,7 +35,14 @@ def analytics
   end
 
   def create_user_event(event_type, user = current_user)
-    Event.create(user_id: user.id, event_type: event_type)
+    Event.transaction do
+      Event.create!(user_id: user.id, event_type: event_type)
+      # Keep total number of events per user under MAX_EVENTS_PER_USER
+      events = Event.where(user_id: user.id).order('created_at ASC')
+      if events.count > Event::MAX_EVENTS_PER_USER
```

Comment:
Though I like using them, callbacks on the model are generally to be avoided, from what I've heard. E.g. http://samuelmullen.com/2013/05/the-problem-with-rails-callbacks/

Could we just make this a daily rake task instead? Like sweeping up old sessions.